DuckDuckGo/TabViewController.swift

```diff
@@ -1146,6 +1151,13 @@ extension TabViewController: WKNavigationDelegate {
             return true
         }
         
+        if let newUrl = linkCleaner.cleanTrackingParameters(initiator: webView.url,
+                                                                   url: navigationAction.request.url) {
+            decisionHandler(.cancel)
+            load(newUrl: newUrl, forNavigationAction: navigationAction)
+            return true
+        }
```

Comment:
Hmm can we have amp link that has tracking parameter? :) If so:
- I wonder how safe is having a multiple 'cancel' decisions.
- Do we have such test case in privacy pages?

Reply:
Also do we have an idea how many urls will be stripped of these links? I'm still concerned about possible impact of cancellation on redirects, reliability and history state of the tab...

Reply:
Since we only clean URLs on deliberate navigation actions (typing URL, link click, open app by url, etc) we don't need to worry about redirects. Additionally history state should be unaffected as these actions happen before URL load.

> Do we have such test case in privacy pages?
I'll add a test case for this but my inclination is there should be no problem. It should follow:
1. AMP link with tracking params is clicked
2. AMP link is rewritten and a new load request is issued
3. AMP link is not detected but tracking parameters are detected and subsequently rewritten.

Reply:
> Do we have such test case in privacy pages?

you can try this URL for the time being and verify the parameter at the end is removed: https://www.google.com/amp/s/www.nytimes.com/2021/09/20/business/jeff-bezos-earth-fund.amp.html?utm_source=test